dll/win32/browseui/shellbars/CBandSiteMenu.cpp

```diff
@@ -27,20 +27,42 @@ CBandSiteMenu::CBandSiteMenu():
     m_hmenu(NULL)
 {
 }
-
 CBandSiteMenu::~CBandSiteMenu()
 {
     if (m_hmenu)
         DestroyMenu(m_hmenu);
 
     if (m_menuDsa)
         DSA_Destroy(m_menuDsa);
+    
+    if (m_DesktopPidl)
+        ILFree(m_DesktopPidl);
+    
+    if (m_QLaunchPidl)
+        ILFree(m_QLaunchPidl);
 
     m_BandSite = NULL;
 }
 
+HRESULT WINAPI CBandSiteMenu::FinalConstruct()
+{
+    HRESULT hr = SHGetFolderLocation(0, CSIDL_DESKTOP, NULL, 0, &m_DesktopPidl);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return hr;
```

Comment:
If you fail here, you'll get an uninitialized `m_QLaunchPidl` in the dtor. It's probably best to initialize both pidls to null in the ctor.

Reply:
Done